pkg/network/discovery.go

```diff
@@ -242,9 +277,9 @@ func (d *DefaultDiscovery) NetworkSize() int {
 
 // updateNetSize updates network size estimation metric. Must be called under read lock.
 func (d *DefaultDiscovery) updateNetSize() {
-	var netsize = len(d.connectedAddrs) + len(d.unconnectedAddrs) + 1 // 1 for the node itself.
-	var fanOut = 2.5 * math.Log(float64(netsize-1))                   // -1 for the number of potential peers.
-	if netsize == 2 {                                                 // log(1) == 0.
+	var netsize = len(d.handshakedAddrs) + len(d.unconnectedAddrs) + 1 // 1 for the node itself.
```

Comment:
Not sure i understand it correctly, if the peer is connected (but not handshakes yet) then it's not in the `handshakedAddrs`, so why don't we take into account connected peers also while calculating network size?

Reply:
Because it might as well be a connection to ourselves, or a duplicate connection, or a bogus node, whatever. Until the handshake is successful it's not a proper network's member.